src/zcash/NoteEncryption.hpp

```diff
@@ -0,0 +1,79 @@
+/*
+See the Zcash protocol specification for more information.
+https://github.com/zcash/zips/blob/master/protocol/protocol.pdf
+*/
+
+#ifndef ZC_NOTE_ENCRYPTION_H_
+#define ZC_NOTE_ENCRYPTION_H_
+
+#include <boost/array.hpp>
+#include "uint256.h"
+
+#include "zerocash/Zerocash.h"
+
+namespace libzcash {
+
+#define NOTEENCRYPTION_AUTH_BYTES 16
```

Comment:
I didn't want to include libsodium in the header file here, so I made this constant and used it in the API, and then statically asserted that it was equal to `crypto_aead_chacha20poly1305_ABYTES` in the cpp file. (This is the same size as `crypto_aead_chacha20poly1305_IETF_ABYTES` and the libsodium maintainers [fixed this problem](https://github.com/jedisct1/libsodium/issues/377) with their documentation.)